src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -660,10 +668,18 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             {
                 BoundExpression argument = rewrittenArguments[a];
                 int p = (!argsToParamsOpt.IsDefault) ? argsToParamsOpt[a] : a;
-                RefKind refKind = argumentRefKinds.RefKinds(a);
-                if (refKind == RefKind.None && parameters[p].RefKind == RefKind.In)
+                RefKind argRefKind = argumentRefKinds.RefKinds(a);
+                RefKind paramRefKind = parameters[p].RefKind;
+
+                // Patch refKinds for arguments that match 'In' parameters to have effective RefKind.
+                // For the purpose of further analysis we will mark the arguments as -
+                // - In       if was originally passed as None
+                // - Ref      if was originally passed as In
```

Comment:
From our discussion, let's try a more formal representation by using a "private enum value" with a good name ("StrictlyIn").

Reply:
replaced with a special internal value of RefKind

---
In reply to: [141971959](https://github.com/dotnet/roslyn/pull/22405#discussion_r141971959) [](ancestors = 141971959)